src/project.ts

```diff
@@ -63,6 +68,24 @@ function examineReleaseVersion(packageVersion: SemVer): ReleaseVersion {
   };
 }
 
+/**
+ * List the workspaces in a monorepo using Yarn.
+ *
+ * @param projectDirectoryPath - The path to the project.
+ * @returns An array of objects that represent the workspaces.
+ */
+export async function listWorkspaces(projectDirectoryPath: string) {
+  const stdout = await getLinesFromCommand(
+    'yarn',
+    ['workspaces', 'list', '--no-private', '--json'],
```

Comment:
Note that I've added `--no-private`. I'm not 100% sure if that's expected, but works for the Snaps repository at least.

Reply:
Actually, we still want to version those. 🤔 I have to re-think this.